mod-source-record-manager-server/src/main/java/org/folio/services/journal/JournalUtil.java

```diff
@@ -70,38 +83,152 @@ record = new ObjectMapper().readValue(recordAsString, Record.class);
         .withActionDate(new Date())
         .withActionStatus(actionStatus);
 
-      if (!isEmpty(entityAsString)) {
-        JsonObject entityJson = new JsonObject(entityAsString);
-        journalRecord.setEntityId(entityJson.getString("id"));
+      if (DI_ERROR == DataImportEventTypes.fromValue(eventPayload.getEventType())) {
+        journalRecord.setError(eventPayloadContext.get(ERROR_KEY));
+      }
 
-        if (entityType == INSTANCE || entityType == HOLDINGS || entityType == ITEM || entityType == PO_LINE) {
+      if ((actionType == JournalRecord.ActionType.MATCH || actionType == JournalRecord.ActionType.NON_MATCH)
+        && (entityType == HOLDINGS || entityType == ITEM)) {
+        List<JournalRecord> resultedJournalRecords = new ArrayList<>();
+        if (isNotBlank(eventPayloadContext.get(NOT_MATCHED_NUMBER))) {
+          int notMatchedNumber = Integer.parseInt(eventPayloadContext.get(NOT_MATCHED_NUMBER));
+          for (int i = 0; i < notMatchedNumber; i++) {
+            resultedJournalRecords.add(constructBlankJournalRecord(record, JournalRecord.ActionType.NON_MATCH, entityType, actionStatus)
+              .withEntityType(entityType)
+              .withEntityId(UUID.randomUUID().toString()));
```

Comment:
propose not to set a random uuid as entityId to journalRecord when holdings/items are not found

Reply:
Thanks, done